File: app/src/main/java/com/example/mobile6/MainDatabase.java
package com.example.mobile6;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.mobile6.dao.ExampleDao;
import com.example.mobile6.dao.MedicineDao;
import com.example.mobile6.model.Example;
import com.example.mobile6.model.Medicine;

@Database(
        entities = {
                Example.class,
                Medicine.class,
        },
        version = 2,
        exportSchema = false
)
public abstract class MainDatabase extends RoomDatabase {
    private static volatile MainDatabase instance = null;

    public abstract ExampleDao exampleDao();
    public abstract MedicineDao medicineDao();

    public static final String DATABASE_NAME = "nhom6.db";

    public static MainDatabase getInstance(Context context) {
        if (instance != null) return instance;
        synchronized (MainDatabase.class) {
            if (instance == null) {
                instance = Room.databaseBuilder(
                                context.getApplicationContext(),
                                MainDatabase.class,
                                DATABASE_NAME
                        )
                        .fallbackToDestructiveMigration()
                        .build();
            }
        }
        return instance;
    }
}

File: app/src/main/java/com/example/mobile6/data/local/room/MainDatabase.kt
package com.example.mobile6.data.local.room

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import com.example.mobile6.data.local.room.converter.DateConverter
import com.example.mobile6.data.local.room.dao.RequestDao
import com.example.mobile6.data.local.room.entity.RequestEntity

@Database(
    entities = [RequestEntity::class],
    version = 1,
    exportSchema = false
)
@TypeConverters(DateConverter::class)
abstract class MainDatabase : RoomDatabase() {
    abstract val requestDao: RequestDao

    companion object {
        const val DATABASE_NAME = "nhom6.db"
    }
}